### ISSUE TITLE
object supporting the buffer API required

### ISSUE THREAD

**Issue by hxueh** [dbcli/mycli#624]
Hi, I have installed mysql@5.7 with homebrew. I use mysql_config_editor to set default [client] login password. When I type `mysql` on terminal it can login MySQL without problem, but when I type `mycli` it will show me `object supporting the buffer API required`. 

But, when I just use regular `mycli -uUser -pPassword` to login it has no problem.

macOS version: 10.13.6
MySQL version: 5.7.22 Homebrew
mycli version: 1.17.0 pip

Below is the log in `.mycli.log`.

```
2018-07-22 01:30:33,368 (48024/MainThread) mycli.main ERROR - traceback: 'Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/mycli/main.py", line 445, in connect
    _connect()
  File "/usr/local/lib/python3.7/site-packages/mycli/main.py", line 398, in _connect
    socket, charset, local_infile, ssl)
  File "/usr/local/lib/python3.7/site-packages/mycli/sqlexecute.py", line 52, in __init__
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/mycli/sqlexecute.py", line 88, in connect
    conv=conv, ssl=ssl)
  File "/usr/local/lib/python3.7/site-packages/pymysql/__init__.py", line 94, in Connect
    return Connection(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pymysql/connections.py", line 327, in __init__
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/pymysql/connections.py", line 598, in connect
    self._request_authentication()
  File "/usr/local/lib/python3.7/site-packages/pymysql/connections.py", line 808, in _request_authentication
    authresp = _auth.scramble_native_password(self.password, self.salt)
  File "/usr/local/lib/python3.7/site-packages/pymysql/_auth.py", line 31, in scramble_native_password
    stage1 = sha1_new(password).digest()
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/hashlib.py", line 149, in __hash_new
    return _hashlib.new(name, data)
TypeError: object supporting the buffer API required
'
```

**Comment by felixsilberstein** [comment]
try passing the password as a string?

**Comment by tsroten** [comment]
@hxueh Thanks for reporting this -- it should be fixed as of yesterday's release, version 1.20.0. You can upgrade if you installed via pip:
`pip install -U mycli`
